### PR TITLE
Update io pystan

### PR DESCRIPTION
**Pull request by madhucharan**
## Description

Related to the issue #1514 , updated the file according to the schema convention

## Checklist

- [x] Follows [official](https://github.com/arviz-devs/arviz/blob/main/CONTRIBUTING.md#pull-request-checklist) PR format
- [ ] Includes a sample plot to visually illustrate the changes (only for plot-related functions)
- [ ] New features are properly documented (with an example if appropriate)?
- [ ] Includes new or updated tests to cover the new feature
- [x] Code style  correct (follows pylint and black guidelines)
- [x] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/main/CHANGELOG.md#v0xx-unreleased)

### CODE REVIEW

**Comment by madhucharan** [comment]
Hi @OriolAbril , I think I have updated the three places, kindly ket me know if there is any changes need to be made

File: arviz/data/io_pystan.py [hunk 1]
```diff
@@ -627,6 +627,15 @@ def get_sample_stats(fit, warmup=False):
     """Extract sample stats from PyStan fit."""
     dtypes = {"divergent__": bool, "n_leapfrog__": np.int64, "treedepth__": np.int64}
 
+    rename_dict = {
+        "divergent": "diverging",
+        "n_leapfrog": "n_steps",
+        "treedepth": "tree_depth",
+        "stepsize": "step_size",
+        "accept_stat": "acceptance_rate",
+        "lp": "lp",
```
Comment (ahartikainen) [inline arviz/data/io_pystan.py:636]:
```suggestion

```
this is not needed

File: arviz/data/io_pystan.py [hunk 2]
```diff
@@ -773,6 +780,15 @@ def get_sample_stats_stan3(fit, variables=None, ignore=None):
     """Extract sample stats from PyStan3 fit."""
     dtypes = {"divergent__": bool, "n_leapfrog__": np.int64, "treedepth__": np.int64}
 
+    rename_dict = {
+        "divergent": "diverging",
+        "n_leapfrog": "n_steps",
+        "treedepth": "tree_depth",
+        "stepsize": "step_size",
+        "accept_stat": "acceptance_rate",
+        "lp": "lp",
```
Comment (ahartikainen) [inline arviz/data/io_pystan.py:789]:
```suggestion

```

same here

File: arviz/data/io_pystan.py [hunk 3]
```diff
@@ -650,20 +659,18 @@ def get_sample_stats(fit, warmup=False):
     data = OrderedDict()
     for key, values in extraction.items():
         values = np.stack(values, axis=0)
-        dtype = dtypes.get(key)
-        values = values.astype(dtype)
+        values = values.astype(dtypes.get(key))
```
Comment (ahartikainen) [inline arviz/data/io_pystan.py:662]: I don't think we need to change these lines

File: arviz/data/io_pystan.py [hunk 4]
```diff
@@ -786,10 +802,9 @@ def get_sample_stats_stan3(fit, variables=None, ignore=None):
         values = fit._draws[fit._parameter_indexes(key)]  # pylint: disable=protected-access
         values = values.reshape(new_shape, order="F")
         values = np.moveaxis(values, [-2, -1], [1, 0])
-        dtype = dtypes.get(key)
-        values = values.astype(dtype)
+        values = values.astype(dtypes.get(key))
```
Comment (ahartikainen) [inline arviz/data/io_pystan.py:805]: Same here, I don't think we need to change these lines.

**Comment by madhucharan** [comment]
@ahartikainen, Updated the changes according to the review.

**Review by ahartikainen — APPROVED** [review]

**Pull request merged**